.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,147 @@
+        env:
+          CATALOG_SOURCE_NAME: "odh-olm-test"
+          CATALOG_SOURCE_NAMESPACE: "olm"
+          SUBSCRIPTION_NAME: "opendatahub-operator"
+          SUBSCRIPTION_NAMESPACE: "openshift-operators"
+
+      - name: Create Namespace for DSPO
+        run: |
+          kubectl create namespace ${{ env.DSPO_NS }}
+          kubectl config set-context --current --namespace=${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Prepare for Upgrade Testing
+        run: |
+          # Update the KfDef manifest with the latest released version
+
+          # Example: Update KfDef manifest to point to the latest released version
+          sed -i "s/main/${{ inputs.released-version }}/" ${{ github.workspace }}/.github/resources-kfdef/kfdef.yaml
+        working-directory: .github/resources-kfdef/
+
+      - name: Print KfDef Manifest Contents
+        run: cat ${{ github.workspace }}/.github/resources-kfdef/kfdef.yaml
+
+      - name: Deploy DSPO KfDef
+        run: |
+          kubectl apply -f .github/resources-kfdef/kfdef.yaml -n ${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: "data-science-pipelines-operator"
+
+      - name: Print ODH Operator Pod Logs
+        run: kubectl get pods -n openshift-operators -o jsonpath='{.items[*].metadata.name}' | xargs -I {} kubectl logs -n openshift-operators {}
+
+      - name: Wait for DSPO deployment to be ready
+        run: |
+          kubectl get deployments -n ${{ env.DSPO_NS }}
+          timeout 300 bash -c 'until [[ $(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}') ]]; do sleep 5 && echo "$(kubectl get deployment/data-science-pipelines-operator-controller-manager -n '${{ env.DSPO_NS }}')"; done'
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Deploy first DSP instance
+        run: |
+          kubectl create namespace ${{ env.DSPA_NS }}
+          cd ${{ env.WORKING_DIR }}/config/samples
+          kustomize build . | kubectl -n ${{ env.DSPA_NS }} apply -f -
+        env:
+          DSPA_NS: test-ds-project-1
+          WORKING_DIR: ${{ github.workspace }}
+
+      - name: Prepare for Upgrade Testing
+        run: |
+          # Update the KfDef manifest with the candidate version
+          sed -i "s/main/${{ inputs.candidate-version }}/" ${{ github.workspace }}/.github/resources-kfdef/kfdef.yaml
+        working-directory: .github/resources-kfdef/
+
+      - name: Print KfDef Manifest Contents
+        run: cat ${{ github.workspace }}/.github/resources-kfdef/kfdef.yaml
+
+      - name: Deploy KfDef Core for the candidate DSP Version
+        run: |
+          kubectl apply -f .github/resources-kfdef/kfdef.yaml -n ${{ env.DSPO_NS }}
+        env:
+          DSPO_NS: data-science-pipelines-operator
+
+      - name: Deploy a DSPA instance for the candidate DSP Version
```

Comment:
Do the same here replace the step and call the new step `Run upgrade tests`.

Reply:
Done, updated to have a separate script and called it here.